Spigot-Server-Patches/0675-Skip-POI-finding-if-stuck-in-vehicle.patch

```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Paul Sauve <paul@technove.co>
+Date: Sat, 20 Feb 2021 16:43:07 +0000
+Subject: [PATCH] Skip POI finding if stuck in vehicle
+
+
+diff --git a/src/main/java/net/minecraft/server/BehaviorFindPosition.java b/src/main/java/net/minecraft/server/BehaviorFindPosition.java
+index 63a761ebef80d4af09cdc2682e496d78492c4a3a..b1cd9d9f263f1525138c35ab818c6cd8117ae57d 100644
+--- a/src/main/java/net/minecraft/server/BehaviorFindPosition.java
++++ b/src/main/java/net/minecraft/server/BehaviorFindPosition.java
+@@ -45,6 +45,12 @@ public class BehaviorFindPosition extends Behavior<EntityCreature> {
+     }
+ 
+     protected boolean a(WorldServer worldserver, EntityCreature entitycreature) {
++        // Paper start - don't calculate POI if they're trapped in a vehicle
++        Entity vehicle = entitycreature.getVehicle();
++        if (vehicle instanceof EntityBoat || vehicle instanceof EntityMinecartRideable) {
```

Comment:
Is there a reason you'd still be looking for POI data when villagers are riding things other than minecarts/boats? Seems like this should just be a `!= null` check?

Reply:
Good point